File: web/src/api/resumeProfiles.ts
import { client } from "./client";
import type {
  ResumeProfile,
  ResumeProfileListResponse,
  ResumeProfileUpdatePayload
} from "../types/resumeProfile";

export async function listResumeProfiles(includeArchived = false): Promise<ResumeProfileListResponse> {
  const response = await client.get<ResumeProfileListResponse>("/api/v1/resume-profiles", {
    params: { include_archived: includeArchived }
  });
  return response.data;
}

export async function getResumeProfile(resumeProfileId: string): Promise<ResumeProfile> {
  const response = await client.get<ResumeProfile>(
    `/api/v1/resume-profiles/${resumeProfileId}`
  );
  return response.data;
}

export async function updateResumeProfile(
  resumeProfileId: string,
  payload: ResumeProfileUpdatePayload
): Promise<ResumeProfile> {
  const response = await client.patch<ResumeProfile>(
    `/api/v1/resume-profiles/${resumeProfileId}`,
    payload
  );
  return response.data;
}

export async function setDefaultResumeProfile(resumeProfileId: string): Promise<ResumeProfile> {
  const response = await client.post<ResumeProfile>(
    `/api/v1/resume-profiles/${resumeProfileId}/default`
  );
  return response.data;
}

export async function archiveResumeProfile(resumeProfileId: string): Promise<ResumeProfile> {
  const response = await client.post<ResumeProfile>(
    `/api/v1/resume-profiles/${resumeProfileId}/archive`
  );
  return response.data;
}

export async function restoreResumeProfile(resumeProfileId: string): Promise<ResumeProfile> {
  const response = await client.post<ResumeProfile>(
    `/api/v1/resume-profiles/${resumeProfileId}/restore`
  );
  return response.data;
}

export async function deleteResumeProfile(resumeProfileId: string): Promise<void> {
  await client.delete(`/api/v1/resume-profiles/${resumeProfileId}`);
}

File: app/api/v1/resume_profiles.py
from __future__ import annotations

from fastapi import APIRouter, Depends, Query, Response, status

from app.api.dependencies import get_current_user
from app.application.resume_profile_usecases import (
    archive_resume_profile,
    delete_resume_profile,
    get_resume_profile,
    list_resume_profiles,
    set_default_resume_profile,
    restore_resume_profile,
    update_resume_profile,
)
from app.schemas.auth import UserAccount
from app.schemas.resume_profile import (
    ResumeProfile,
    ResumeProfileListResponse,
    ResumeProfileUpdateRequest,
)

router = APIRouter(prefix="/api/v1/resume-profiles", tags=["v4-resume-profiles"])


@router.get("", response_model=ResumeProfileListResponse)
def list_resume_profiles_endpoint(
    include_archived: bool = Query(default=False),
    current_user: UserAccount = Depends(get_current_user),
) -> ResumeProfileListResponse:
    return ResumeProfileListResponse(
        items=list_resume_profiles(
            user_id=current_user.user_id,
            include_archived=include_archived,
        )
    )


@router.get("/{resume_profile_id}", response_model=ResumeProfile)
def get_resume_profile_endpoint(
    resume_profile_id: str,
    current_user: UserAccount = Depends(get_current_user),
) -> ResumeProfile:
    return get_resume_profile(resume_profile_id, user_id=current_user.user_id)


@router.patch("/{resume_profile_id}", response_model=ResumeProfile)
def update_resume_profile_endpoint(
    resume_profile_id: str,
    payload: ResumeProfileUpdateRequest,
    current_user: UserAccount = Depends(get_current_user),
) -> ResumeProfile:
    return update_resume_profile(
        resume_profile_id,
        payload,
        user_id=current_user.user_id,
    )


@router.post("/{resume_profile_id}/default", response_model=ResumeProfile)
def set_default_resume_profile_endpoint(
    resume_profile_id: str,
    current_user: UserAccount = Depends(get_current_user),
) -> ResumeProfile:
    return set_default_resume_profile(resume_profile_id, user_id=current_user.user_id)


@router.post("/{resume_profile_id}/archive", response_model=ResumeProfile)
def archive_resume_profile_endpoint(
    resume_profile_id: str,
    current_user: UserAccount = Depends(get_current_user),
) -> ResumeProfile:
    return archive_resume_profile(resume_profile_id, user_id=current_user.user_id)


@router.post("/{resume_profile_id}/restore", response_model=ResumeProfile)
def restore_resume_profile_endpoint(
    resume_profile_id: str,
    current_user: UserAccount = Depends(get_current_user),
) -> ResumeProfile:
    return restore_resume_profile(resume_profile_id, user_id=current_user.user_id)


@router.delete("/{resume_profile_id}", status_code=status.HTTP_204_NO_CONTENT)
def delete_resume_profile_endpoint(
    resume_profile_id: str,
    current_user: UserAccount = Depends(get_current_user),
) -> Response:
    delete_resume_profile(resume_profile_id, user_id=current_user.user_id)
    return Response(status_code=status.HTTP_204_NO_CONTENT)
